docs/administration/configuration/configuring-tls-certificates/index.md

```diff
@@ -0,0 +1,66 @@
+# Configuring TLS certificates
+
+## Default self-signed certificate
+
+A self-signed certificate is automatically generated each time REANA's Helm
+chart is deployed or upgraded. This certificate is valid for 90 days and it is
+stored in a Kubernetes secret named `<helm-release-prefix>-tls-secret`. To
+generate a new TLS certificate, you can simply perform an upgrade of your REANA
+instance using `helm upgrade`.
+
+!!! warning
+    When using `helm upgrade`, remember to specify all the values you
+    have set when deploying the chart and the version of REANA you would like to
+    use, in case you do not want to upgrade to the latest version.
+
+You can disable the generation of this certificate by setting
+`ingress.tls.self_signed_cert` to `false`. If you want to use longer lasting
+certificates, see the [Using a custom certificate](#using-a-custom-certificate)
+section.
+
+## Using a custom certificate
+
+It is possible to configure REANA to use an already existing certificate, which
+can either be a valid certificate issued by a trusted CA, or a custom
+self-signed one. Please note that a self-signed certificate will not be
+considered valid by browsers, unless users explicitly trust it.
+
+You can create a self-signed TLS certificate for `localhost` with `openssl`:
+
+```{ .console .copy-to-clipboard }
+$ # source https://letsencrypt.org/docs/certificates-for-localhost/
+$ openssl req -x509 -out cert.crt -keyout cert.key \
+  -newkey rsa:2048 -nodes -sha256 -days 365 \
+  -subj '/CN=localhost' -extensions EXT -config <( \
+   printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth")
+```
+
+This command will create two files: `cert.crt` and `cert.key`, containing
+respectively the public and private part of the certificate. The certificate
+will last 365 days, but you can change its duration by modifying the value of
+the `-days` flag.
+
+You can then create a
+[Kubernetes TLS secret](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets)
+containing the certificate, which in this example will be called
+`reana-custom-tls-secret`:
+
+```{ .console .copy-to-clipboard }
+$ kubectl create secret tls reana-custom-tls-secret --cert=./cert.crt --key=./cert.key
+```
+
+Finally, you must set the necessary Helm values:
+
+- `ingress.tls.self_signed_cert` should be set to `false`, so that the default
+  self-signed certificate is not generated
+- `ingress.tls.secret_name` should be set to the name of the Kubernetes secret
+  containing the certificate, in this case `reana-custom-tls-secret`
+
+## Automatic certificate issuance
+
+Depending on how your cluster is configured, you might be able to let Kubernetes
+handle the issuance and configuration of TLS certificates. Different ingress
+controllers require different configurations, but sometimes you might need to
```

Comment:
This section reads a bit cryptic. I guess we would have to provide some Let's Encrypt tentative example to make it clearer to the reader.  Something like:

You may also be able to let Kubernetes handle the issuance and configuration of TLS certificates so that the process would be fully automatic. Different ingress controllers may require different configurations, but the setup usually requires to annotate the ingress object. For example, in order to use the Let's Encrypt certificate service with Traefik ingress controller, it would benecessary to discover and edit the ingress config map:

```
$ # discover ingress 
$ kubectl get configmaps --all-namespaces | grep ingress
$ # edit config map to input the acme section as listed below
$ kubectl edit configmaps -n kube-system ingress-traefik
$ # delete ingress pod to trigger certificate issuance 
$ kubectl -n kube-system delete $(kubectl -n kube-system get pod -o name | grep traefik)
```

where the newly added acme section reads as follows:

```
    [acme]
      email = "john.doe@example.org"
      storage = "acme.json"
      entryPoint = "https"
      ACMELogging = true
    [acme.tlsChallenge]
    [[acme.domains]]
      main = "reana.example.org"
```